audio.go

```diff
@@ -0,0 +1,351 @@
+package kvm
+
+import (
+	"io"
+	"sync"
+	"sync/atomic"
+
+	"github.com/jetkvm/kvm/internal/audio"
+	"github.com/jetkvm/kvm/internal/logging"
+	"github.com/pion/webrtc/v4"
+	"github.com/rs/zerolog"
+)
+
+const (
+	socketPathOutput = "/var/run/audio_output.sock"
+	socketPathInput  = "/var/run/audio_input.sock"
+)
+
+var (
+	audioMutex           sync.Mutex
+	outputSupervisor     *audio.Supervisor
+	inputSupervisor      *audio.Supervisor
+	outputClient         *audio.IPCClient
+	inputClient          *audio.IPCClient
+	outputRelay          *audio.OutputRelay
+	inputRelay           *audio.InputRelay
+	audioInitialized     bool
+	activeConnections    atomic.Int32
+	audioLogger          zerolog.Logger
+	currentAudioTrack    *webrtc.TrackLocalStaticSample
+	inputTrackHandling   atomic.Bool
+	useUSBForAudioOutput bool
+	audioOutputEnabled   atomic.Bool
+	audioInputEnabled    atomic.Bool
+)
+
+func initAudio() {
+	audioLogger = logging.GetDefaultLogger().With().Str("component", "audio-manager").Logger()
+
+	if err := audio.ExtractEmbeddedBinaries(); err != nil {
+		audioLogger.Error().Err(err).Msg("Failed to extract audio binaries")
+		return
+	}
+
+	// Load audio output source from config
+	ensureConfigLoaded()
+	useUSBForAudioOutput = config.AudioOutputSource == "usb"
+
+	// Enable both by default
+	audioOutputEnabled.Store(true)
+	audioInputEnabled.Store(true)
+
+	audioLogger.Debug().
+		Str("source", config.AudioOutputSource).
+		Msg("Audio subsystem initialized")
+	audioInitialized = true
+}
+
+// startAudioSubprocesses starts audio subprocesses and relays (skips already running ones)
+func startAudioSubprocesses() error {
+	audioMutex.Lock()
+	defer audioMutex.Unlock()
+
+	if !audioInitialized {
+		audioLogger.Warn().Msg("Audio not initialized, skipping subprocess start")
+		return nil
+	}
+
+	// Start output subprocess if not running and enabled
+	if outputSupervisor == nil && audioOutputEnabled.Load() {
+		alsaDevice := "hw:0,0" // HDMI
+		if useUSBForAudioOutput {
+			alsaDevice = "hw:1,0" // USB
+		}
+
+		outputSupervisor = audio.NewSupervisor(
+			"audio-output",
+			audio.GetAudioOutputBinaryPath(),
+			socketPathOutput,
+			[]string{
+				"ALSA_CAPTURE_DEVICE=" + alsaDevice,
```

Comment:
Aren't these already set to the correct values in the C source?

Reply:
I thought of just using the defaults too... initially I wanted to leave the ability to alter these parameters on the fly, but there's really no need. Thanks!